cpp/ql/lib/semmle/code/cpp/security/PrivateData.qll

```diff
@@ -0,0 +1,63 @@
+/**
+ * Provides classes for heuristically identifying variables and functions that
+ * might contain or return sensitive private data.
+ *
+ * 'Private' data in general is anything that would compromise user privacy if
+ * exposed. This library tries to guess where private data may either be stored
+ * in a variable or returned by a function call.
+ *
+ * This library is not concerned with credentials. See `SensitiveExprs.qll` for
+ * expressions related to credentials.
+ */
+
+import cpp
+
+/** A string for `match` that identifies strings that look like they represent private data. */
+private string privateNames() {
+  result =
+    [
+      // Inspired by the list on https://cwe.mitre.org/data/definitions/359.html
+      // Government identifiers, such as Social Security Numbers
+      "%social%security%",
+      // Contact information, such as home addresses and telephone numbers
+      "%postcode%", "%zipcode%", "%telephone%",
```

Comment:
Since we already allow arbitrary strings between the word `bank` and `account`, how about we also allow arbitrary strings between `post` and `code`, and `zip` and `code`?

Our tests would improve if we added `%licese%key%` as well (or maybe just `%license%`?). Do we catch more good results if we add those?

(These issues aren't at all blocking. Feel free to postpone them to another PR if you agree that this is something we should do)

Reply:
> Since we already allow arbitrary strings between the word bank and account, how about we also allow arbitrary strings between post and code, and zip and code?

I did some test runs on LGTM and this turned out to be problematic.  `post`, `code` and `zip` are commonly used in variable and function names referring to technical things, so we get a few too many false positives.

I may try switching from `match` to regexps, and allow exactly 0 or 1 arbitrary character in between words in all of these cases - rather than doing the best we can with the options of any string (often too permissive) or nothing (too restrictive).

> Our tests would improve if we added %licese%key% as well (or maybe just %license%?). Do we catch more good results if we add those?

Planned as a follow-up, as I think that is closer to a password than private personal data so should be a change in `SensitiveExprs.qll` rather than `PrivateData.qll`.

Reply:
I've now addressed your first point, the second will be addressed in a follow-up PR.

Reply:
Follow-up PR here: https://github.com/github/codeql/pull/8623